server/src/main/java/org/elasticsearch/cluster/routing/IndexShardRoutingTable.java

```diff
@@ -477,6 +471,15 @@ public ShardRouting getByAllocationId(String allocationId) {
     }
 
     public Set<String> getAllAllocationIds() {
+        Set<String> allAllocationIds = new HashSet<>();
+        for (ShardRouting shard : shards) {
```

Comment:
The master node doesn't use this field anyway ... if we want faster here on the data node I have an open PR for that  https://github.com/elastic/elasticsearch/pull/79837 :)

Reply:
Can we assert that it is not called on the master thread with a suitable comment? That could document this optimization.

Reply:
Sure, added that assertion.